File: c/Shell_Sort.c
#include <stdio.h>  
using namespace std;
#define MAX 100
void shell_sort(int arr[], int num)    
{    
      int i, j, k, tmp;    
          for (i = num / 2; i > 0; i = i / 2)    
            {    
               for (j = i; j < num; j++)    
                  {    
                     for(k = j - i; k >= 0; k = k - i)    
                        {    
                          if (arr[k+i] >= arr[k])    
                                 break;    
                          else    
                            {    
                               tmp = arr[k];    
                               arr[k] = arr[k+i];    
                               arr[k+i] = tmp;    
                             }    
                          }    
                    }    
                }    
}    

int main()    
{    
int arr[MAX];    
int k,  num;    
printf("Enter total no. of elements : ");    
scanf("%d", &num);    
printf("\nEnter %d numbers: ", num);    
  
          for (k =  0 ; k < num; k++)    
                  scanf("%d", &arr[k]);    
    
shellsort(arr, num);    
printf("\n Sorted array is: ");    
             for (k = 0; k < num; k++)    
                       printf("%d ", arr[k]);    
return 0;    
}    

File: rust/shell_sort.rs
fn shell_sort<T: Ord>(arr: &mut [T]) {
    for i in 0..arr.len() {
        for j in 0..arr.len() - 1 - i {
            if arr[j] > arr[j + 1] {
                arr.swap(j, j + 1);
            }
        }
    }
}

fn main() {
    let mut array = vec![0, 8, 4, 7, 1, 9, 10, 5, 2, 11, 3];
    println!("{:?}", array);
    shell_sort(&mut array);
    println!("{:?}", array);

}